subprojects/core/src/integTest/groovy/org/gradle/api/services/BuildServiceIntegrationTest.groovy

```diff
@@ -71,33 +71,62 @@ class BuildServiceIntegrationTest extends AbstractIntegrationSpec {
         succeeds 'broken'
     }
 
-    private taskUsingUndeclaredService() {
+    def "does not nag when service is used by task with an explicit usesService call and feature preview is enabled"() {
+        given:
+        serviceImplementation()
+        customTaskUsingServiceViaProperty()
         buildFile """
-            def provider = gradle.sharedServices.registerIfAbsent("counter", CountingService) {
+            def serviceProvider = gradle.sharedServices.registerIfAbsent("counter", CountingService) {
                 parameters.initial = 42
             }
 
-            tasks.register("broken") {
+            tasks.register("explicit") {
+                it.usesService(serviceProvider)
                 doFirst {
-                    provider.get().increment()
+                    serviceProvider.get().increment()
                 }
             }
         """
+        settingsFile '''
+            enableFeaturePreview 'STABLE_CONFIGURATION_CACHE'
+        '''
+
+        when:
+        succeeds 'explicit'
+
+        then:
+        outputDoesNotContain "'Task#usesService'"
     }
 
-    def "service is created once per build on first use and stopped at the end of the build"() {
+    def "does not nag when service is annotated with @ServiceReference and feature preview is enabled"() {
+        given:
         serviceImplementation()
+        customTaskUsingServiceViaProperty("@${ServiceReference.class.name}(name = 'foobar')")
```

Comment:
❓ Wouldn't `name = 'foobar'` have to match a registered service name?

Reply:
Note that this PR, although declares a `name` property on the ServiceReference annotation, does not care about the value provided. 

This other PR implemented that:  https://github.com/gradle/gradle/pull/21754/files#diff-177b01df4f530a32fd73f07c48f1767fb6ce34465477be561e6a71f163434f36R104